marshaler_example_test.go

```diff
@@ -62,7 +62,7 @@ func ExampleMarshaler() {
 		},
 	}
 
-	logger := zap.New(zap.NewJSONEncoder(zap.NoTime()))
+	logger := zap.New(zap.WriterFacility(zap.NewJSONEncoder(zap.NoTime()), nil, zap.InfoLevel))
```

Comment:
IMHO this doesn't really seem like an improvement to the API.

How about leaving zap.New the same as it is, as that's what most clients will probably want, and add a new function that can create a logger from a facility when more control is required?